src/Stratis.Bitcoin.Features.Miner/BlockDefinition.cs

```diff
@@ -259,50 +272,46 @@ protected void OnBuild(ChainedHeader chainTip, Script scriptPubKey)
             //LogPrint(BCLog::BENCH, "CreateNewBlock() packages: %.2fms (%d packages, %d updated descendants), validity: %.2fms (total %.2fms)\n", 0.001 * (nTime1 - nTimeStart), nPackagesSelected, nDescendantsUpdated, 0.001 * (nTime2 - nTime1), 0.001 * (nTime2 - nTimeStart));
         }
 
-        // Add a tx to the block.
-        private void AddToBlock(TxMempoolEntry iter)
+        /// <summary>
+        /// Adds a transaction to the block from the given mempool entry.
+        /// </summary>
+        private void AddToBlock(TxMempoolEntry mempoolEntry)
         {
-            this.logger.LogTrace("({0}.{1}:'{2}')", nameof(iter), nameof(iter.TransactionHash), iter.TransactionHash);
+            this.logger.LogTrace("({0}.{1}:'{2}')", nameof(mempoolEntry), nameof(mempoolEntry.TransactionHash), mempoolEntry.TransactionHash);
 
-            this.block.AddTransaction(iter.Transaction);
+            this.block.AddTransaction(mempoolEntry.Transaction);
 
-            this.BlockTemplate.VTxFees.Add(iter.Fee);
-            this.BlockTemplate.TxSigOpsCost.Add(iter.SigOpCost);
+            this.BlockTemplate.VTxFees.Add(mempoolEntry.Fee);
+            this.BlockTemplate.TxSigOpsCost.Add(mempoolEntry.SigOpCost);
 
             if (this.NeedSizeAccounting)
-                this.BlockSize += iter.Transaction.GetSerializedSize();
+                this.BlockSize += mempoolEntry.Transaction.GetSerializedSize();
 
-            this.BlockWeight += iter.TxWeight;
+            this.BlockWeight += mempoolEntry.TxWeight;
             this.BlockTx++;
-            this.BlockSigOpsCost += iter.SigOpCost;
-            this.fees += iter.Fee;
-            this.inBlock.Add(iter);
-
-            //bool fPrintPriority = GetBoolArg("-printpriority", DEFAULT_PRINTPRIORITY);
-            //if (fPrintPriority)
-            //{
-            // LogPrintf("fee %s txid %s\n",
-            //  CFeeRate(iter->GetModifiedFee(), iter->GetTxSize()).ToString(),
```

Comment:
I suggest we debug/trace this line? we should have access to the fee right?

Reply:
I've added this to the trace log at the start of the method :)